lib/web_ui/lib/src/engine/text/ruler.dart

```diff
@@ -405,6 +405,10 @@ class ParagraphRuler {
       ..border = '0'
       ..padding = '0';
 
+    if (assertionsEnabled) {
+      _singleLineHost.setAttribute('data-ruler', 'single-line');
```

Comment:
What does "data-" in the attribute name mean?

Reply:
Nothing. It's just for debugging purposes. When I debug, I sometimes make ruler elements visible to see how they look like and what their sizes are. It's easier to have this attribute than trying to guess based on `white-space: pre`, etc.

It's an old convention to use the `data-` prefix with user-defined attributes.